src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2024,7 +2022,7 @@ private ICaseClauseOperation CreateBoundPatternSwitchLabelOperation(BoundPattern
             else if (boundPatternSwitchLabel.WhenClause == null &&
                      boundPatternSwitchLabel.Pattern.Kind == BoundKind.ConstantPattern &&
                      boundPatternSwitchLabel.Pattern is BoundConstantPattern cp &&
-                     cp.Value.Type.IsValidV6SwitchGoverningType())
+                     cp.InputType.IsValidV6SwitchGoverningType())
```

Comment:
>cp.InputType [](start = 21, length = 12)

Please add a regular IOperation tree test (not a flow graph test) for a scenario that makes this change observable.